packages/api/test/scripts/containers.js

```diff
@@ -0,0 +1,111 @@
+import path from 'path'
+import fs from 'fs'
+import { Client as Minio } from 'minio'
+import pg from 'pg'
+import { fileURLToPath } from 'url'
+import { DockerComposeEnvironment, Wait } from 'testcontainers'
+
+const __dirname = fileURLToPath(new URL('.', import.meta.url))
+const composeDir = path.join(__dirname, '..', '..', 'docker')
+const schemaDir = path.join(__dirname, '..', '..', 'db')
+
+export async function startTestContainers() {
+  // setting COMPOSE_COMPATIBILITY = true
+  // forces docker-compose v2 to use names with underscores, e.g. db_1
+  // and means we don't need to care which version is in use
+  process.env.COMPOSE_COMPATIBILITY = 'true'
+  const compose = await new DockerComposeEnvironment(
+    composeDir,
+    'docker-compose.yml'
+  )
+    // testcontainers has an issue with the default port-based healthcheck
+    // that seems related to https://github.com/testcontainers/testcontainers-node/issues/40
+    // using log-message wait strategy avoids this
+    .withWaitStrategy('rest_1', Wait.forLogMessage('Listening on port'))
+    .withWaitStrategy(
+      'db_1',
+      Wait.forLogMessage('PostgreSQL init process complete')
+    )
+    .withWaitStrategy('ipfs_1', Wait.forLogMessage('Daemon is ready'))
+    .withWaitStrategy('cluster_1', Wait.forLogMessage('IPFS Cluster is READY'))
+    .withWaitStrategy('minio_1', Wait.forLogMessage('1 Online'))
+    .up()
+
+  const ports = {
+    minio: compose.getContainer('minio_1').getMappedPort(9000),
```

Comment:
what does getMappedPort mean ? is it dynamic or not ? is the 9000 just like "try 9000 if that fail give me a free port" ?

Reply:
Give me the docker container port mapped to the internal service running on 9000

Reply:
so, for the host port it is dynamic yes, the 9000 is defined in docker compose for the service inside the container

Reply:
got it thank you